corehq/apps/app_manager/util.py

```diff
@@ -276,6 +277,11 @@ def get_properties(self, case_type, already_visited=(),
                 case_properties.update(p for p in updates if "/" not in p)
             case_properties.update(self.get_save_to_case_updates(form, case_type))
 
+        if toggles.DATA_DICTIONARY.enabled(self.app.domain):
+            data_dict_props = CaseProperty.objects.filter(case_type__domain=self.app.domain,
+                                                          case_type__name=case_type)
+            case_properties |= {prop.name for prop in data_dict_props}
```

Comment:
Would it be better/faster to get the case type hierarchy from the `ParentPropertyBuilder` (and not the case properties) and only get the properties from the data dictionary